packages/react/src/components/TooltipIcon/TooltipIcon.js

```diff
@@ -57,7 +75,9 @@ const TooltipIcon = ({
       className={tooltipTriggerClasses}
       aria-describedby={tooltipId}
       onMouseEnter={composeEventHandlers([onMouseEnter, handleMouseEnter])}
-      onFocus={composeEventHandlers([onFocus, handleFocus])}>
+      onMouseLeave={composeEventHandlers([onMouseLeave, handleMouseLeave])}
+      onFocus={composeEventHandlers([onFocus, handleMouseEnter])}
```

Comment:
Added this to the `mouseEnter` block, but let me know if we want to handle this separately